File: project_1_ETL_HEALTHCARE/frontend/src/api/analytics.ts
import axios from "axios";

import type {
  MonthlyAdmission,
  BedUtilization,
  DiseaseTrend,
  LabWorkload,
} from "../types/analytics";

const API = "http://127.0.0.1:8000";

export async function getMonthlyAdmissions(): Promise<MonthlyAdmission[]> {
  const res = await axios.get(`${API}/analytics/monthly-admissions`);
  return res.data;
}

export async function getBedUtilization(): Promise<BedUtilization[]> {
  const res = await axios.get(`${API}/analytics/bed-utilization`);
  return res.data;
}

export async function getDiseaseTrends(): Promise<DiseaseTrend[]> {
  const res = await axios.get(`${API}/analytics/disease-trends`);
  return res.data;
}

export async function getLabWorkload(): Promise<LabWorkload[]> {
  const res = await axios.get(`${API}/analytics/lab-workload`);
  return res.data;
}

File: project_1_ETL_HEALTHCARE/backend/app/api/analytics.py
from fastapi import APIRouter

from app.services.analytics_service import AnalyticsService

router = APIRouter()


@router.get("/monthly-admissions")
def monthly_admissions():
    return AnalyticsService.monthly_admissions()


@router.get("/bed-utilization")
def bed_utilization():
    return AnalyticsService.bed_utilization()


@router.get("/disease-trends")
def disease_trends():
    return AnalyticsService.disease_trends()


@router.get("/lab-workload")
def lab_workload():
    return AnalyticsService.lab_workload()


@router.get("/inventory")
def inventory():
    return AnalyticsService.medication_inventory()